grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -172,18 +177,24 @@
         this.unsafeWrapResponseBuffers = unsafeWrapResponseBuffers;
         grpcWebText = GrpcSerializationFormats.isGrpcWebText(serializationFormat);
         this.maxInboundMessageSizeBytes = maxInboundMessageSizeBytes;
-        endpointInitialized = endpointGroup instanceof Endpoint || endpointGroup instanceof StaticEndpointGroup;
```

Comment:
I think this was an optimization? It's probably not entirely true to say that an endpoint is initialized given that in some cases we never call `ctx.init(endpoint)`: when compressor isn't found (this was the case before), when deadline is exceeded (new case).

Reply:
That is correct. We changed the logic that initiates the endpoint to using `initContextAndExecuteWithFallback` a while ago so I guess this wasn't valid anymore from that time. 
Thanks for noticing it.